### PR TITLE
Fix `declaration-block-no-redundant-longhand-properties`/`declaration-block-no-shorthand-property-overrides` false negatives for `font-variant`

### PR DESCRIPTION
**Pull request by Bilie**
> Which issue, if any, is this issue related to?

Closes https://github.com/stylelint/stylelint/issues/7620

> Is there anything in the PR that needs further explanation?

Hi, I have updated the properties file and added few tests, let me know if there is anything missing or can be improved

### CODE REVIEW

**Review by ybiquitous — CHANGES_REQUESTED** [review]
@Bilie Thanks for the pull request! Can you address my review comments?

---

[suggest] Can you remove the following `skip` in `declaration-block-no-shorthand-property-overrides` test cases? It should pass now.

https://github.com/stylelint/stylelint/blob/b5b3799cc107ede5b418bef6ef64bab072b2deab/lib/rules/declaration-block-no-shorthand-property-overrides/__tests__/index.mjs#L211

File: lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs [hunk 1]
```diff
@@ -923,6 +923,77 @@ testRule({
...
+						font-variant-position: foo;
+						font-variant-caps: foo;
+						font-variant-numeric: foo;
+						font-variant-alternates: foo;
+						font-variant-east-asian: foo;
+						font-variant-emoji: foo;
+						font-variant: foo;
+					}
+				`,
+			message: messages.expected('font-variant'),
```
[the remark this answers is not in the document]
Comment (ybiquitous) [reply lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:954]: In addition, to test if a fixed code is in the correct order, can you replace only `foo` values with other values (e.g. `foo1`, `foo2`, etc.)?

File: lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs [hunk 2]
```diff
@@ -923,6 +923,77 @@ testRule({
...
+			code: stripIndent`
+					a {	
+						font-variant-ligatures: foo;
+						font-variant-position: foo;
+						font-variant-caps: foo;
+						font-variant-numeric: foo;
+						font-variant-alternates: foo;
+						font-variant-east-asian: foo;
+						font-variant-emoji: foo;
+						font-variant: foo;
```
Comment (Mouvedia) [inline lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:951]: medium

Could you put real values?
When all values are `foo` the test cannot detect that the order is incorrect from the `fixed` string.
This is partly my fault.

**Comment by Bilie** [comment]
Hi thank you both for the review, it has been really helpful! 😊 I think I have addressed all comments, but if I have missed anything, just let me know

File: lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs [hunk 3]
```diff
@@ -68,6 +68,13 @@ testRule({
 			code: 'a { background-repeat: repeat; background-attachment: scroll; background-position: 0% 0%; background-color: transparent; background-image: none; background-origin: border-box; background-clip: text; }',
 			description: 'missing background-size',
 		},
+		{
+			code: stripIndent`
+					a {
+						font-variant: foo1 foo2 foo3 foo4 foo5 foo6 foo7;
+					}
+				`,
```
Comment (ybiquitous) [inline lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:76]: [suggest] Can you keep indentation consistent with other test cases?

```suggestion
			code: stripIndent`
				a {
					font-variant: foo1 foo2 foo3 foo4 foo5 foo6 foo7;
				}
			`,
```

File: lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs [hunk 4]
```diff
@@ -861,6 +868,25 @@ testRule({
...
+					a {
+						font-variant-ligatures: foo1;
+						font-variant-position: foo2;
+						font-variant-caps: foo3;
+						font-variant-numeric: foo4;
+						font-variant-alternates: foo5;
+						font-variant-east-asian: foo6;
+						font-variant-emoji: foo7;
+					}
+				`,
```
Comment (ybiquitous) [inline lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:882]: [suggest] Let's keep indentation consistent.

```suggestion
			code: stripIndent`
				a {
					font-variant-ligatures: foo1;
					font-variant-position: foo2;
					font-variant-caps: foo3;
					font-variant-numeric: foo4;
					font-variant-alternates: foo5;
					font-variant-east-asian: foo6;
					font-variant-emoji: foo7;
				}
			`,
```

File: lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs [hunk 5]
```diff
@@ -861,6 +868,25 @@ testRule({
...
+			message: messages.expected('font-variant'),
+			fixed: stripIndent`
+				a {
+					font-variant: foo1 foo2 foo3 foo4 foo5 foo6 foo7;
+				}
+			`,
```
Comment (ybiquitous) [inline lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:888]: [suggest] Let's reorder `fixed` and `message` like other cases:

```suggestion
			fixed: stripIndent`
				a {
					font-variant: foo1 foo2 foo3 foo4 foo5 foo6 foo7;
				}
			`,
			message: messages.expected('font-variant'),
```

File: lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs [hunk 6]
```diff
@@ -923,6 +949,28 @@ testRule({
 	],
 });
 
+testRule({
+	ruleName,
+	config: [true, { ignoreShorthands: ['/^font-variant/'] }],
```
Comment (ybiquitous) [inline lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:954]: [suggest] `ignoreShorthands` has been tested already, so can you remove this case?

Ref https://github.com/stylelint/stylelint/pull/7734/files#diff-1f86a8d8167e5474d3acb984827780d6f4fdc4b66d36478c75280f22c7e2b3cbR976

File: lib/rules/declaration-block-no-shorthand-property-overrides/__tests__/index.mjs [hunk 7]
```diff
@@ -208,7 +208,6 @@ testRule({
 			message: messages.rejected('font-variant', 'font-variant-caps'),
 			line: 1,
 			endLine: 1,
```
Comment (ybiquitous) [inline lib/rules/declaration-block-no-shorthand-property-overrides/__tests__/index.mjs:210]: [suggest] In addition, it'd be great if you could add `column`/`endColumn` like other cases 🙏🏼 

```suggestion
			line: 1,
			column: 36
			endLine: 1,
			endColumn: 48
```

**Review by ybiquitous — CHANGES_REQUESTED** [review]
@Bilie Thanks! Can you take a look at my additional comments, please? This PR seems overall good, though.

**Comment by Mouvedia** [comment]
off topic

It seems like there's a typo on https://developer.mozilla.org/en-US/docs/Web/CSS/font-variant.
It should be `Specifies the keywords and functions related to the font-variant-ALTERNATES longhand property.`.

**Review by ybiquitous — APPROVED** [review]
Thank you. LGTM 👍🏼

**Pull request merged**